diff --git a/docroot/includes/common.inc b/docroot/includes/common.inc
index 09c3d5a6..cd1cec08 100644
--- a/docroot/includes/common.inc
+++ b/docroot/includes/common.inc
@@ -611,8 +611,9 @@ function drupal_parse_url($url) {
   }
   // The 'q' parameter contains the path of the current page if clean URLs are
   // disabled. It overrides the 'path' of the URL when present, even if clean
-  // URLs are enabled, due to how Apache rewriting rules work.
-  if (isset($options['query']['q'])) {
+  // URLs are enabled, due to how Apache rewriting rules work. The path
+  // parameter must be a string.
+  if (isset($options['query']['q']) && is_string($options['query']['q'])) {
     $options['path'] = $options['query']['q'];
     unset($options['query']['q']);
   }
@@ -5551,6 +5552,7 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
   // there in favor of sites/all or sites/<domain> directories.
   $profiles = array();
   $profile = drupal_get_profile();
+  
   // For SimpleTest to be able to test modules packaged together with a
   // distribution we need to include the profile of the parent site (in which
   // test runs are triggered).
