diff --git a/docroot/includes/common.inc b/docroot/includes/common.inc
index 34bd4edf..09c3d5a6 100644
--- a/docroot/includes/common.inc
+++ b/docroot/includes/common.inc
@@ -5560,14 +5560,6 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
       $profiles[] = $testing_profile;
     }
   }
-  // In case both profile directories contain the same extension, the actual
-  // profile always has precedence.
-  $profiles[] = $profile;
-  foreach ($profiles as $profile) {
-    if (file_exists("profiles/$profile/$directory")) {
-      $searchdir[] = "profiles/$profile/$directory";
-    }
-  }
 
   // Always search sites/all/* as well as the global directories.
   $searchdir[] = 'sites/all/' . $directory;
@@ -5580,6 +5572,16 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
   if (!function_exists('file_scan_directory')) {
     require_once DRUPAL_ROOT . '/includes/file.inc';
   }
+
+  // In case both profile directories contain the same extension, the actual
+  // profile always has precedence.
+  $profiles[] = $profile;
+  foreach ($profiles as $profile) {
+    if (file_exists("profiles/$profile/$directory")) {
+      $searchdir[] = "profiles/$profile/$directory";
+    }
+  }
+
   foreach ($searchdir as $dir) {
     $files_to_add = file_scan_directory($dir, $mask, array('key' => $key, 'min_depth' => $min_depth));
 
